pmda/parallel.py

```diff
@@ -121,7 +121,32 @@ def _conclude(self):
                # sensible new variable.
                self.results = np.hstack(self._results)
                # Apply normalisation and averaging to results here if wanted.
-               self.results /= np.sum(self.results)
+               self.results /= np.sum(self.results
+
+           @staticmethod
+           def _reduce(res, result_single_frame):
+               # NOT REQUIRED
+               # Called for every frame. ``res`` contains all the results
+               # before current time step, and ``result_single_frame`` is
+               # the result of self._single_frame for the current time step.
+               # Here the single frame result is `added` to ``res``. Return
+               # the new ``res``.
+               # One can define the way to `add` result for a single frame.
+               # The default is to generate a python list, and append the
+               # result for a single frame to the list. We call this kind of
+               # ``res`` `timeseries`.
+               res.append(result_single_frame)
+               # If one wants ``self._reduce`` to sum up results for each
+               # single frame, there's one example in `rdf.py`. We call it
+               # `accumulator`.
+               if res == []:
+               # Convert res from an empty list to a numpy array which has the
+               # same shape as the single frame result
+                   res = result_single_frame
+               else:
+               # Add two numpy arrays
+                   res += result_single_frame
+               return res
```

Comment:
Thanks for adding the documentation. Here is my suggestion for an updated version. I use your structure but try to make the text read more fluently. 

```python
               # NOT REQUIRED
               # Called for every frame. ``res`` contains all the results
               # before current time step, and ``result_single_frame`` is the
               # result of self._single_frame for the current time step. The
               # return value is the updated ``res``. The default is to append
               # results to a python list. This approach is sufficient for
               # time-series data.
               res.append(results_single_frame)
               # This is not suitable for every analysis. To add results over
               # multiple frames this function can be overwritten. The default
               # value for ``res`` is an empty list. Here we change the type to
               # the return type of `self._single_frame`. Afterwards we can
               # safely use addition to accumulate the results.
               if res == []:
                   res = result_single_frame
               else:
                   res += result_single_frame
               # If you overwrite this function *always* return the updated
               # ``res`` at the end.
               return res

```

Reply:
A reference to `_reduce` also needs to be added [here](https://github.com/MDAnalysis/pmda/blob/44984e05a7852ddc5d5aae7053f1cec95638c3a2/pmda/parallel.py#L98)

Reply:
```
To parallelize the analysis ``ParallelAnalysisBase`` separates the trajectory
into work blocks containing multiple frames. The number of blocks is equal to
the number of available cores or dask workers. This minimizes the number of
python processes that are started during a calculation. Accumulation of frames within a block happens in the `self._reduce` function.
 A consequence when
using dask is that adding additional workers during a computation will not
result in an reduction of run-time. 
```
I think it would be beneficial to add this text at the beginning of the class documentation to better explain why `_reduce` is needed.